gitapp/lib/user_details/user_details_api_repository.dart

```diff
@@ -0,0 +1,89 @@
+import 'dart:convert';
+import 'package:gitapp/user_details/user_details_model.dart';
+import 'package:gitapp/user_details/user_repo_model.dart';
+import "package:http/http.dart" as http;
+
+late int page;
+
+
+class ReposList {
+
+  List<UserRepo> listRepos;
+
+  ReposList({required this.listRepos});
+
+  factory ReposList.fromJson(List<dynamic> json){
+    List<UserRepo> listRepos = <UserRepo>[];
+    listRepos = json.map((r) => UserRepo.fromJSON(r)).toList();
+    return ReposList(listRepos: listRepos);
+  }
+
+}
+
+Future<ReposList> getReposListFromAPI(int userId) async {
+
+  page = 1;
```

Comment:
Are we bumping the page number in some place?

Reply:
Yes, we are. When fetching more repos from API.